src/VocabBuilder.scala

```diff
@@ -1,227 +0,0 @@
-//package cc.factorie.app.nlp.embeddings
-import cc.factorie.util.Threading
-import cc.factorie.app._
-import scala.collection.mutable.ArrayBuffer
-import scala.math
-import scala.util.{Sorting, Random}
-import java.util.zip.{GZIPInputStream, GZIPOutputStream}
-import java.io.{OutputStreamWriter, BufferedOutputStream, FileInputStream, FileOutputStream}
-
-
-class VocabBuilder(vocab_hash_size: Int = 20e6.toInt, sampling_table_size: Int = 1e8.toInt, load_factor: Double = 0.7) {
-
-  // Highly Optimized hash table using linear probing 
-  private var vocab_max_size = 1000
-  private var vocab_size = 0
-  private var min_reduce = 1
-  private val power: Double = 0.75
-
-  // assign internal params 
-  private val rng = new Random(5) // fix the seed
-  private var train_words = 0
-  private var prev_vocab_size = 0
-
-  // allocate data structures
-  private var vocab_hash = new Array[Int](vocab_hash_size) // vocab_hash stores pointer to the vocab array 
-  private var vocab = new Array[vocab_word](vocab_max_size) //  vocab array holds the (word, cnt) pairs. In future, huffman encoding will be added. 
```

Comment:
All my reading suggests these are stored in this format but I'm not sure how/if they get converted to one hot before feed forward and I'm also curious at what point word sense vectors aka v(s)(w(t),k) get converted to one hot as well (see Figure 2).